### PR TITLE
[bug] NoUnusedImportsFixer - import not removed when class name is part of constant

### PR DESCRIPTION
**Pull request by kubawerlos**
When I remove dash it will work fine.

### CODE REVIEW

**Comment by SpacePossum** [comment]
Hi,

could you add this to the test:

```php
<?php
use A\ABC;
$a = 5-ABC::Test;
```

**Comment by kubawerlos** [comment]
@SpacePossum not sure why, but done.

**Comment by SpacePossum** [comment]
Ah sorry, combined these pass.
Lemme try again;

```php
    /**
     * @param string $expected
     *
     * @dataProvider provideWhenImportedClassIsUsedForConstantsCases
     */
    public function testWhenImportedClassIsUsedForConstants($expected)
    {
        $this->doTest($expected);
    }

    public function provideWhenImportedClassIsUsedForConstantsCases()
    {
        return array(
            array(
                '<?php
use A\ABC;
$a = 5-ABC::Test;
$a = 5-ABC::Test-5;
$a = ABC::Test-5;
'
            ),
            array(
'<?php
use A\ABC;
$a = 5-ABC::Test;
$a = 5-ABC::Test-5;
'
            ),
            array(
                '<?php
use A\ABC;
$a = 5-ABC::Test;
'
            ),
            array('<?php
use A\ABC;
$a = ABC::Test-5;
'
            ),
            array('<?php
use A\ABC;
$a = 5-ABC::Test-5;
'
            ),
        );
    }
```

These result in 3 failures which I think needs looking into.

**Comment by kubawerlos** [comment]
I have added yet one more case.

**Comment by SpacePossum** [comment]
thanks :)

I start to wonder though, can this be reliable done? And if so, is the `-` exception the only one to consider (what about `$a = 'TEST+a;'`) ?

ping @julienfalque IIRC you had some ideas for this fixer as well.

**Comment by julienfalque** [comment]
IMO using regular expressions **cannot** be reliable enough. This PR is good as it improves the fixer, but I still think we should rewrite it anyway. I'll try to take some time to implement https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/2814#issuecomment-307545175 soon.

**Comment by keradus** [comment]
so, @julienfalque , @SpacePossum , please decide is there anything left to do in this PR or it's RTM ?

**Comment by SpacePossum** [comment]
I think the proposed regex solution is not going to be reliable, if others feel confident than I'm happy to re-review.

File: tests/Fixer/Import/NoUnusedImportsFixerTest.php [hunk 1]
```diff
@@ -684,6 +684,136 @@ public function testWithSameNamespaceImportAndUnusedImportAfterNamespaceStatemen
...
+',
+            ),
+            array('<?php
+use A\ABC;
+$a = 5-ABC::Test-5;
+',
+            ),
+            array('<?php
+use A\ABC;
+$b = $a-->ABC::Test;
```
Comment (julienfalque) [inline tests/Fixer/Import/NoUnusedImportsFixerTest.php]: I'm surprised this is valid PHP, what does this line do exactly?

Comment (SpacePossum) [reply tests/Fixer/Import/NoUnusedImportsFixerTest.php]: `$b` will be a boolean, the value is resolved by `$a--` than `>` than `ABC::Test`,
or throwing in some stuff to make it more easy;
`$b = $a-->ABC::Test;`
===
`$b = ($a--) > ABC::Test;`

Comment (julienfalque) [reply tests/Fixer/Import/NoUnusedImportsFixerTest.php]: Thanks :)

**Comment by julienfalque** [comment]
I'm closing this PR as it is included in #3445, thanks!

**Pull request closed without merging**